Add initial page for important myst syntax. (#542)
GeodynamicWorldBuilder/WorldBuilder: doc/sphinx/developer_manual/contributing_to_the_documentation/important_syntax.md

Important syntax
===============

The world builder is using [sphinx](https://www.sphinx-doc.org) to generatate the documentation. Sphinx accepts multiple varients of extended markdown, but in this project we are using [myst](https://myst-parser.readthedocs.io). Below we will list some commen commands to use.

:::{note}
Everything in this file will look like garbage in a regular markdown viewer, like if you’re viewing this on github. Viewing it on readthedocs will render everything properly.
:::

(title)=
# Title
Every page starts with a header for this file:
::::{code-block} md
(part:dev_manual:chap:contribute_to_doc:sec:important_syntax)=
Important syntax
=====================================

::::

> Note that the first line contains a path to the file so that it can be linked from the other files.

Reviewer: ```suggestion
Note that the first line contains a reference to this file that can be used to link to this file from other places of the documentation.
```

(text)=
# Text

Put single asterisks (*) around text you *want to italicize*, but double asterisks around text you **want to bold**. If you want to `highlight code` you put single Grave Accent (\`) around it. You can escape chacters with the backlash (\), like so \`not highlighed\`, \*not italic\*, \*\*not bold\*\*.

:::{code-block} md
Put single asterisks (*) around text you *want to italicize*, 
but double asterisks around text you **want to bold**. 
If you want to `highlight code` you put single Grave Accent (\`) around it.

> You can escape chacters with the backlash (\), like so \`not highlighed\`, \*not italic\*, \*\*not bold\*\*.

Reviewer: ```suggestion
You can escape chacters with the backlash (\\), like so \`not highlighed\`, \*not italic\*, \*\*not bold\*\*.
```

:::

(headers)=
# Headers

Headers start with a # for level 1, ## for level 2, etc. 

(header2)=
## Header 2

(header3)=
### Header 3

:::{code-block} md
(header1)=
# Header 1

(header2)=
## Header 2

(header3)=
### Header 3
:::

# Links
Links are written in the following way:
:::{code-block} md
[link within this page](title)
:::
[link within this page](title)

:::{code-block} md
[link to other page](part:dev_manual:chap:contribute_to_doc:sec:index)
:::

[link to other page](part:dev_manual:chap:contribute_to_doc:sec:index)

:::{code-block} md
[link to external website](https://github.com/GeodynamicWorldBuilder/WorldBuilder)
:::
[link to external website](https://github.com/GeodynamicWorldBuilder/WorldBuilder)

# Admonitions

> Adding note, warning or and todo's which stand out can be done with the following syntax:

Reviewer: ```suggestion
Adding notes, warnings or and todos which stand out can be done with the following syntax:
```

:::

::::{code-block} md
:::{attention}
This is an `attention` admonition.
:::
::::
:::{attention}
This is an `attention` admonition.
:::

::::{code-block} md
:::{danger}
This is a `danger` admonition.
:::
::::
:::{danger}
This is a `danger` admonition.
:::

::::{code-block} md
:::{error}
This is an `error` admonition.
:::
::::
:::{error}
This is an `error` admonition.
:::

# Figures

Adding figures looks similair to notes and todo's:

::::{code-block} md
:::{figure} ../../../../doc/sphinx/_static/images/user_manual/basic_starter_tutorial/BST_17.png
:name: BST_17_synatx
:alt: Basic Starter Tutorial section 17 highres result. 
:align: center

Basic Starter Tutorial section 17 high resolution result. This has 8 times the resolution than the grid file above. Note that you can see the Earth's curvature! 
:::
::::

:::{figure} ../../../../doc/sphinx/_static/images/user_manual/basic_starter_tutorial/BST_17.png
:name: BST_17_synatx
:alt: Basic Starter Tutorial section 17 highres result. 
:align: center

Basic Starter Tutorial section 17 high resolution result. This has 8 times the resolution than the grid file above. Note that you can see the Earth's curvature! 
:::

# Code blocks

> Making code blocks is similair to making notes or todos'. Instead of `{note}` you need to use `{code-block}` and then the name of the language which the code should be colored by. For example for json, you would use:

Reviewer: ```suggestion
Highlighting code blocks is similar to making notes or todos. Instead of `{note}` you need to use `{code-block}` and then the name of the language which the code should be colored by. For example for json, you would use:
```

::::{code-block} md
:::{code-block} json
:lineno-start: 1
:emphasize-lines: 3,4
 {
    "version":0.6.0,
    "coordinate system": {"model":"cartesian"}, 
 }
:::
::::

> The lines starting with a colon (:) specifies certain parameters which can be set, wuch as setting line number and what number they need to start at (lineno-start: 1) and what lines to mark (emphasize-lines: 3,4).

Reviewer: ```suggestion
The lines starting with a colon (:) specify certain parameters which can be set, such as showing line numbers and what number they need to start at (lineno-start: 1) and what lines to mark (emphasize-lines: 3,4).
```

:::{code-block} json
:lineno-start: 1
:emphasize-lines: 3,4
 {
    "version":0.6.0,
    "coordinate system": {"model":"cartesian"}, 
    "Features":[]
 }
:::

If you need to wrap a code block in a code block, you just use more : on the ouside:

:::::{code-block} md
::::{code-block} md
:::{code-block} json
:lineno-start: 1
:emphasize-lines: 3,4
 {
    "version":0.6.0,
    "coordinate system": {"model":"cartesian"}, 
 }
:::
::::
:::::

Including code from files can be done in the following way:
::::{code-block} md
:::{literalinclude} ../../_static/gwb_input_files/BST_02_minimal_box.wb
:language: json
:lineno-start: 1
:::
::::

:::{literalinclude} ../../_static/gwb_input_files/BST_02_minimal_box.wb
:language: json
:lineno-start: 1
:::

# Dropdown
:::::{code-block}

> ::::{dropdown} Click me to drop down

Reviewer: ```suggestion
::::{dropdown} Click me to open a dropdown box
```

:name: this:is:my:label

> This text is only show when you click on the dropdown button.

Reviewer: ```suggestion
This text is only shown when you click on the dropdown button.
```

::::
:::::

> ::::{dropdown} Click me to drop down

Reviewer: ```suggestion
::::{dropdown} Click me to open a dropdown box
```

:name: this:is:my:label

> This text is only show when you click on the dropdown button.

Reviewer: ```suggestion
This text is only shown when you click on the dropdown button.
```

::::

# Grids

Girds work through multiple layers, like a code-block in a code block:

:::::{code-block}
::::{grid} 3
:::{grid-item-card} BST_17_spherical_models.wb
:link: ../../_static/gwb_input_files/BST_17_spherical_models.wb
:::
:::{grid-item-card} BST_17_spherical_models.grid
:link: ../../_static/gwb_input_files/BST_17_spherical_models.grid
:::
:::{grid-item-card} Paraview Spherical state file 
:link: ../../_static/paraview_state_files/BST_spherical.pvsm
:::
::::
:::::

::::{grid} 3
:::{grid-item-card} BST_17_spherical_models.wb
:link: ../../_static/gwb_input_files/BST_17_spherical_models.wb
:::
:::{grid-item-card} BST_17_spherical_models.grid
:link: ../../_static/gwb_input_files/BST_17_spherical_models.grid
:::
:::{grid-item-card} Paraview Spherical state file 
:link: ../../_static/paraview_state_files/BST_spherical.pvsm
:::
::::

# Tabs

:::::{code-block}

::::{tab-set}
:::{tab-item} tab 1
:sync: tab1
 tab 1
:::

:::{tab-item} tab 2
:sync: tab2

> tab2

Reviewer: ```suggestion
tab 2
```

:::
::::
:::::

::::{tab-set}
:::{tab-item} tab 1
:sync: tab1
 tab 1
:::

:::{tab-item} tab 2
:sync: tab2

> tab2

Reviewer: ```suggestion
tab 2
```

:::
::::

# Citations
:::{code-block} md
Traditional citation {cite}`Fraters_Thieulot_etal_2019`.
Citation as noun {cite:t}`Fraters_Thieulot_etal_2019`.
Use a comma for multiple, like {cite:t}`Fraters_Thieulot_etal_2019,Kronbichler_Heister_etal_2012,Heister_Dannberg_etal_2017`.
:::

becomes

Traditional citation {cite}`Fraters_Thieulot_etal_2019`.
Citation as noun {cite:t}`Fraters_Thieulot_etal_2019`.
Use a comma for multiple, like ({cite:t}`Fraters_Thieulot_etal_2019,Kronbichler_Heister_etal_2012,Heister_Dannberg_etal_2017`.

If the paper you wish to cite is not already in [references.bib](https://github.com/geodynamics/aspect/blob/main/doc/sphinx/bibliography.bib), you will need to add its bibtex entry. Please format the citation tag as:

- `firstauthor:year` for one author
- `firstauthor:secondauthor:year` for two authors
- `firstauthor:secondauthor:year` for three authors

> - `firstauthor:secondauthor:etal:year` for more than two authors

Reviewer: this format is different from your current format. Want to update?
